### DISCUSSION TITLE
Volantis 7.0 意见征集

### DISCUSSION THREAD

**Discussion by xaoxuu** [volantis-x/hexo-theme-volantis#818] [Ideas]
- 您认为 `Volantis` 相较于其他主题最大的优点是什么？
- 您认为 `Volantis` 相较于其他主题最大的缺点是什么？
- 您最希望 `7.0` 拥有的新特性或者改变是什么？

欢迎在下方留下您的宝贵意见，并进行投票吧～

**Comment by penndu** [comment]
@xaoxuu 准备回归Volantis的主题建设了吗？

优点：

1. 模块丰富，主流功能都有；
2. 团队积极，提出的问题都会及时满足及处理；
3. 更新较快，几乎每周都有新鲜代码提交。


缺点：

1. 相对代码臃肿，不知道是个人关系还是主题原因，站点生成速度巨慢；
2. 特色功能较少，希望具备一些咱主题独特的功能，或是效果；
3. 不算缺点，平台支持单一，如果像Next主题支持多种博客架构……

**Reply by xaoxuu** [reply]
是的，接下来有时间就开始设计了

**Comment by BeaCox** [comment]
+ 优点：
   1. 社区活跃，问题反馈比较及时
   2. 可自定义性、可扩展性强，可玩性高
   3. 大部分人认为必要的功能都有，还有标签插件等比较实用而且许多主题并不具备的功能
   
+ 缺点：
  1. 文档更新不能及时跟上主题更新
  2. 不同版本之间配置项（以及标签写法）有较大差异，希望统一风格
  3. 主题样式缺乏新意

6.0希望迎来样式的改变。另外希望侧边栏可以增加自定义部件的功能（像butterfly那样），虽然现在可以用通用页面部件替代，但通过这种方式自定义感觉还是有些...
感谢各位开发者大佬贡献！😍

**Reply by xaoxuu** [reply]
- 标签写法以前一直在探索不同的格式，现在通过 Stellar 的尝试基本确定了最合理的格式，后续重构之后应该不需要再改了。
- 样式之前一直在 Material-Flow 的基础上进行调整和扩展的，下个版本将会彻底重新设计了。

**Reply by penndu** [reply]
> * 标签写法以前一直在探索不同的格式，现在通过 Stellar 的尝试基本确定了最合理的格式，后续重构之后应该不需要再改了。
> * 样式之前一直在 Material-Flow 的基础上进行调整和扩展的，下个版本将会彻底重新设计了。

这是不是意味着之前文章里使用的标签格式需要调整呢？

**Reply by xaoxuu** [reply]
是的，应该是最后一次了，Stellar 用了快两年了，这个格式非常完美。

**Reply by easyf12** [reply]
是否考虑将Stellar的功能和Volantis的美观集合在一起呢？

**Reply by xaoxuu** [reply]
@easyf12 正在做

**Reply by talefou** [reply]
大佬现在还打算继续开发嘛 看看 [@且听风吟](https://stellar.listentothewind.cn) 博主的修改呗 完美将 Stellar 和 Volantis 结合在了一起

**Reply by Tikas** [reply]
不知是不是我开了低功耗的原因，访问且听风吟超卡

**Comment by xaoxuu** [comment]
现代化风格的设计
![首页 - 个人博客布局](https://user-images.githubusercontent.com/16400144/195126140-e09e8411-ec91-43eb-a500-c68ae00228a0.png)
![首页 - 社区布局](https://user-images.githubusercontent.com/16400144/195126153-98ece030-75a3-4030-8c85-aadce55fd848.png)
![首页 - 文章列表布局](https://user-images.githubusercontent.com/16400144/195126161-1b638166-57d7-4460-adf0-4dcff5529366.png)
![首页 - 项目列表+文章列表布局](https://user-images.githubusercontent.com/16400144/195126168-4d8df1f1-f170-4aea-9fd0-f4c985803d11.png)
![文档 - 上](https://user-images.githubusercontent.com/16400144/195126172-eee54951-cd2f-4616-99e5-0d9f523dcb12.png)
![文章 - 上](https://user-images.githubusercontent.com/16400144/195126177-ad83a457-10a5-43b5-adea-6c685fd57fcc.png)
![项目列表](https://user-images.githubusercontent.com/16400144/195126181-0601db44-4afd-43cd-84d7-61a306de0e73.png)

**Reply by penndu** [reply]
看效果图，新版貌似是单栏的，可否添加双栏模式，因为有些信息需要展示！

**Reply by xaoxuu** [reply]
左右两边都可以有侧边栏

**Reply by inkss** [reply]
好棒！！！

**Reply by 289373410** [reply]
这个很牛

**Comment by xaoxuu** [comment]
降低上手难度，简化配置，删减掉一些实用性不高又难以维护的功能，例如（请留言告知）

> 我个人是很想去掉pjax的

**Reply by BeaCox** [reply]
个人感觉`parallax`实用性不高。因为如果`position`为`cover`，看见效果的概率很小，而`position`为`fixed`作为背景可能影响阅读体验。而且还有一点比较鸡肋的是，现在通过配置文件为volantis配置背景图片不得不引入`parallax.js`。

**Reply by BeaCox** [reply]
> 我个人是很想去掉pjax的

糟糕...我倒挺中意pjax...

**Reply by penndu** [reply]
> > 我个人是很想去掉pjax的
> 
> 糟糕...我倒挺中意pjax...

pjax确实会影响一些冲突函数调用。

**Comment by Tikas** [comment]
### 优点

+ 1、拥有一个非常活跃的社区，提供强大的支援帮助。
+ 2、创新的思维，使 volantis 位列时尚风。
+ 3、多功能，让 volantis 拥有灵活的适用性。
+ 4、最主要的是开发者们个个男的帅到爆，女的美到令人心动就算了，他 / 她 们为人还热情、善良、体贴等等，做事又能提供多方案解决，并能择优再优化方案，做事能力强 ... ... 此处省略 1000000 个字 ... ... 此段描述结束！

### 缺点

+ 1、当启用过多的功能时，导致整个页面代码非常臃肿，剑指 cpu 处理性能的考核。
+ 2、由于主题非常时尚，css 代码量也是非常多，不知可否按需加载，具体表达如：
  -  2.1、头部、尾部、整体风格（如适配不同屏幕）的代码，可以归为核心 css，每个页面都加载，此 css 文件适用 **通用 cdn** 加速，如果此文件不大，同样适合直接集成到 html 代码中，以减少请求。
  -  2.2、个性化页面的 css ，为此页面特有的，只此类页面加载，比如一些【说说】、【特别功能】、【BB】之类的页面，此类页面可以归为功能类，和下面结合一起。
  -  2.3、相关非核心的功能模块，可以再划出一个功能 css 代码，不同的用户，选择不同功能，此功能 css 文件，会不相同，此文件，**不适用** 通用 cdn 加速，当然，也要提供一个全功能的此 css 文件 cdn 加速。此类功能如：音乐播放器等功能
>这样就每个页面只需要加载核心 css 文件，和功能类 css 文件，如果用户追求基本功能的话，那么第二个功能类 css 文件就不会出现使用场景。
+ 3、多风格，是一大优点，同样也产生了更多的 css 相关代码，如果主题的核心 css 文件，能以用户使用不同的风格页面，替换此风格的核心 css 文件，可以进一步减少代码的臃肿。

*这些是来自一个一直忙于工作，css 还未学完，整天想努力成为开发者的不专业发言。*

### 非常喜欢此卡片风格，如移动端显示的外观

![image](https://user-images.githubusercontent.com/3396710/195225144-64a7038f-8633-4271-9dfd-043a2726bfb8.png)

**Reply by xaoxuu** [reply]
感谢这么丰富的建议，个性化页面有在考虑，目前主要在 Stellar 上用标签方式试水，把各种功能做成基础标签就可以在任何页面任何位置用这些特殊功能，不同标签组合起来实现更丰富的自由度。
例如把 `tabs` 标签简化成了无边框设计，把 `noteblock` 标签简化成了只有边框和底色，这样两者组合起来使用就相当于 `tabs` 多了一种样式（再x10种颜色）。

**Reply by Tikas** [reply]
我期待学有所成时，正式发布新的主题以丰富 volantis ，我的目标，让 volantis 成为 hexo 里非常出名的主题系列，而不是单一主题，XDD，还是先忙完工作吧

**Reply by xaoxuu** [reply]
期待 👏

**Comment by xaoxuu** [comment]
**文章检索栏目为**
```
最新发布 | 最近更新 | 分类 | 标签 | 归档 | 朋友文章
```
其中「朋友文章」就是「友链朋友圈」，但是想以更好的方式实现和集成，最好是以 GitHub 为服务提供方。
此外，还想把 Stellar 上已经写好的基于 Issues 数据源的时间线也放在合适的地方显示，目前倾向于做成和「文章」、「项目」并列的一个层级「动态」。

> 分类和标签更适合放在搜索页

**Reply by penndu** [reply]
文章检索栏目是导航吗？希望导航可自定义！
![image](https://user-images.githubusercontent.com/46226528/195271328-ae4340f4-66b8-49e8-8afe-bce56273174a.png)
此处增加显示分类：默认、最新发布「隐藏置顶」最新更新「按updata排序」归档、朋友文章。
附议分类和标签更适合放搜索页。

**Reply by xaoxuu** [reply]
参考一下 Stellar 的文章列表页面和文章页面，按照目前的设计想法，如果左右两边都支持侧边栏，顶部就不需要再加顶栏了：
- [文章列表页面](https://xaoxuu.com/)：滚动时文章检索栏目会自动吸顶，因为这时候是在找文章，可能随时会需要切换检索方式
- [文章阅读页面](https://xaoxuu.com/blog/2020-09-27-heartmate/)：有左右侧边栏可以存放足够多的自定义内容，顶部没有存在的意义了（原标题+导航栏，导航栏可在侧边显示，标题不始终显示在顶部也可以的）

至于置顶，计划放在文章检索栏上面的区域，以横向滚动或拼图方式呈现，类似这样：
<img width="1017" alt="image" src="https://user-images.githubusercontent.com/16400144/195294784-fe54836f-bb37-4064-8930-5f00598738c3.png">

**Reply by penndu** [reply]
相当于将顶部导航栏移到了页面左侧！

**Reply by xaoxuu** [reply]
是的，滚动时左侧固定，右侧跟随内容滚动，可以根据需要两边放不同的内容

**Reply by penndu** [reply]
> 是的，滚动时左侧固定，右侧跟随内容滚动，可以根据需要两边放不同的内容

你说的越具体，我就越想催着你赶紧做~

**Reply by BeaCox** [reply]
> > 是的，滚动时左侧固定，右侧跟随内容滚动，可以根据需要两边放不同的内容
> 
> 你说的越具体，我就越想催着你赶紧做~

俺也一样！

**Comment by imjason** [comment]
优点:
1.结构简洁，功能俱全
2.团队活跃
3.其他的大家都说过了
缺点
1.性能不佳希望优化
建议
1之前的TODO
优化多作者模式，归档页面模仿[极光](https://github.com/auroral-ui/hexo-theme-aurora) *修正链接*
2页脚和个人介绍可以学习[zhheo](blog.zhheo.com)
3增加侧边工具栏类似bf的
4以及我之前提的issue把弹窗和音乐播放改成灵动岛
5其他想起来再说吧

**Reply by xaoxuu** [reply]
极光链接404了

**Reply by Tikas** [reply]
不是 404，是少了一个 r，原本是 auroral，写成 auroal
正确地址：
https://github.com/auroral-ui/hexo-theme-aurora

**Reply by xaoxuu** [reply]
好的，会结合一下各家的好创意

**Comment by MCSeekeri** [comment]
我个人是很希望Vlts能够兼容别的静态博客生成器的，因为我站点文字量巨大，wordcount结果是1901k字，每次生成网站都得等个好几分钟……
我也曾考虑过换到Hugo，但是鉴于我大量运用了Vlts的标签……还是放弃了

除此之外还有就是感觉代码过于冗余，app.js 有点太大了……之前有人提出的拆分按需加载我就很支持。

**Reply by xaoxuu** [reply]
大佬有没有用Action自动部署啊

**Reply by MCSeekeri** [reply]
> 大佬有没有用Action自动部署啊

用了，这里的数据就是看Action日志得到的。
仓库地址[在此](https://github.com/MCSeekeri/sciadv)，可能是我单文章太长了才会生成这么慢……
不过其实有了Action之后生成速度慢这个问题不是很碍事，但是我要是想本地边改边看结果就很难受。

**Comment by imjason** [comment]
还有一个建议，如果可以把说说或者短文反正首页夹杂在文章中间那会很棒，如果可以调用第三方API就更好了，例如qexo说说
ispeak等等

**Reply by Tikas** [reply]
XDDD，其实我一直学 CSS ，就是想自己弄个多种风格的文章排列，这个是异次元给了我很大感触。

**Reply by xaoxuu** [reply]
会支持多种布局的摘要卡片，但是本地文章和API穿插显示不太好。

**Comment by 289373410** [comment]
优点不多说。
说说几个小建议，仅代表个人建议，无任何攻击性。
1.代码确实有点冗余，有一些样式是固定死的，对于我这种后端开发，前端半吊子的人来说，可读性不高。如打赏模块，它在文章这个ejs底部，并没有做分离，以至于我想自定义一个新的打赏样式，就会牵一发而动全身，会出现摁下葫芦起了瓢，是不是可以支持多种css语法呢。

2.头部有个消失的动画，也就是调整头部透明度的，这个值目前为0，如果切换一个明亮的背景，会闪一下才消失。

3.配置项里有许多图片配置，如果不配置，如作者头像，那么文章里就会预留出这个图片的位置，可不可以不设置，这个位置就不做预留呢。

**Reply by xaoxuu** [reply]
这些重构之后应该都可以优化掉的

**Reply by Tikas** [reply]
多来些痛点，好样的，毕竟，开发时，可能并没有考虑到这些使用场景，而留下了痛点，感谢反馈！

**Comment by xaoxuu** [comment]
[风记星辰](https://www.thyuu.com/) 这个大佬的博客太漂亮了

**Reply by Tikas** [reply]
进  awwwards  了吗

**Reply by xaoxuu** [reply]
> 进 awwwards 了吗

这是什么意思

**Reply by xaoxuu** [reply]
打算模仿到关于页

**Reply by Tikas** [reply]
https://www.awwwards.com/

**Reply by Tikas** [reply]
我经常来这看别人做的网站

**Reply by xaoxuu** [reply]
看了些感觉都不合适

**Reply by Tikas** [reply]
是的，多是一些酷炫的，并不是所有的都是博客类，目前我也想找一个博客类的来亮亮眼，好像这个还是国人做的好

**Reply by THYUU** [reply]
> 看了些感觉都不合适

是的，因为国外的网站都是根据英文字母来进行排版设计，可是我们中国的汉字并不适合他们的风格。你会察觉英文字母在设计中无论横板或者竖版真的很好看，并不是说汉字丑。

一方面是因为方块字在排版中比较死板，甚至还有人们母语的习惯认知，造成我们过度解读汉字后产生的廉价感。另一方面是当你好好去欣赏繁体汉字后，就会发现要比简体汉字好看，中国为了减少文盲，牺牲并优化了繁杂的汉字，留下的产物就是现在的简体字。再加上中国汉字的巅峰一直都是书法，而非现在的计算机黑体。

**Reply by xaoxuu** [reply]
大佬来了！🥰

但是还是得用简体，不然读者看不懂。非计算机行业博主可以选择用一些艺术点的字体，计算机类一般还是得用普普通通的字体，不然跟代码放一起就很不搭了。

**Reply by Tikas** [reply]
萌生了我想弄个以汉字排版好看的主题，哈哈，先来这丢个坑

**Comment by penndu** [comment]
来自Volantis忠实用户的问候，新版本的进展如何，今年还有望发布吗？

**Reply by xaoxuu** [reply]
今年可能性不大了，到年底都没什么时间 😭

**Reply by Tikas** [reply]
来自Volantis忠实用户的问候，我也在努力学习主题开发，目前 css 的运用已经能初步运用，js 依旧还未学习，丰富 Volantis 系列主题，让 Volantis 成为 hexo 里成名的主题系列今年可能性不大了，到年底都没什么时间😭

**Reply by Tikas** [reply]
今天难得打开了久违的 github，就又再次把目光集中在主题开发。

正如我在 hexo-theme-stellar 里回复的那样准备使用 Astro，Hexo 的主题开发，总感觉代码会冗余很多，很奇妙，我并不是一个成熟的前端人员，相反，我只是一个刚学 CSS 的人，很多时候都需要一边查阅相关知识才能配合主题的开发。

然而 Hexo 给我的感觉就是不够高效，很多资源无法做到合理的利用，在停滞开发（主要也是年底在忙），就一直寻找新的解决方案。

这时，Astro 进入了我的眼中。

其实已经有一段时间了，只是一直没能抽空去具体了解，加上我的 JS 没有学习，主题开发的道路，还很远。

就在此时，我竟然看到了这篇文章，深有同感，我想充实 volantis 的新主题，可以是 Astro

[Hexo 的表演该落幕了，让它退场吧](https://stblog.penclub.club/posts/removeHexo/)

**Reply by SeqCrafter** [reply]
> 今天难得打开了久违的 github，就又再次把目光集中在主题开发。
> 
> 正如我在 hexo-theme-stellar 里回复的那样准备使用 Astro，Hexo 的主题开发，总感觉代码会冗余很多，很奇妙，我并不是一个成熟的前端人员，相反，我只是一个刚学 CSS 的人，很多时候都需要一边查阅相关知识才能配合主题的开发。
> 
> 然而 Hexo 给我的感觉就是不够高效，很多资源无法做到合理的利用，在停滞开发（主要也是年底在忙），就一直寻找新的解决方案。
> 
> 这时，Astro 进入了我的眼中。
> 
> 其实已经有一段时间了，只是一直没能抽空去具体了解，加上我的 JS 没有学习，主题开发的道路，还很远。
> 
> 就在此时，我竟然看到了这篇文章，深有同感，我想充实 volantis 的新主题，可以是 Astro
> 
> [Hexo 的表演该落幕了，让它退场吧](https://stblog.penclub.club/posts/removeHexo/)

偶然间点开了你发的这个博客链接，里面的内容简直令人感到震惊！！！

**Reply by Tikas** [reply]
这没什么吧，哈哈，我刚用 hexo 就感觉很不良好，当时是在用着 vuepress，正如我的回复那样，现在，我在使用 Astro 开发 Volantis-go 主题，这个框架有点大，加上业余开发，期望年底能有个初版本上线，感兴趣可以关注一下吧！

**Reply by SeqCrafter** [reply]
> 这没什么吧，哈哈，我刚用 hexo 就感觉很不良好，当时是在用着 vuepress，正如我的回复那样，现在，我在使用 Astro 开发 Volantis-go 主题，这个框架有点大，加上业余开发，期望年底能有个初版本上线，感兴趣可以关注一下吧！

不是，我说其他的博文，男-->女

**Reply by Tikas** [reply]
哈哈，好吧，我不会在这讨论这些，每个人都有自己的权利，嘻嘻，低调低调，技术不分性别，我会尊重每个人

**Comment by xingpingcn** [comment]
6.0的话我希望可以适配md3风格，因为我很喜欢那个配色，而且chrome从118（还是117）开始也能够根据系统主题动态更换浏览器的配色了，如果博客也能调用浏览器api同步更换颜色就更酷了。

**Reply by xaoxuu** [reply]
是 `#2196f3` 吗

**Reply by xingpingcn** [reply]
我指的是[这种](https://www.ithome.com/0/647/227.htm)。如果网页开发不像安卓开发一样很方便就能适配md3风格，那也不必为此增加写代码的难度
